### PR TITLE
#147 - Major application configuration overhaul

### PR DESCRIPTION
**Pull request by plencka**
Most parts of the application were either moved, deleted or adjusted, which results with a lot of changes.

Important stuff that was done:
- Updated Docker containers
- Configured Laravel, which is contained in the `backend` folder
  - Laravel path configuration had to be adjusted in various places like bootstrap, AppServiceProvider.
- Moved year old backend to the `backend/static` to make it separate from Laravel, it's going to be removed later.
- Removed vue cli in favor of vite (with laravel and vue plugins) for frontend building.
  - Development environment does not run properly, but assets can be compiled for local web server.
- The frontend seems to work correctly without many changes (~~excluding the MapBox map, which will be investigated in next pull requests~~)

This pull request is done with intention to make next ones smaller. :slightly_smiling_face:

### CODE REVIEW

**Comment by sbanasiak** [comment]
MapBox map should be fixed now

File: frontend/views/app.blade.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<html class="h-full bg-blumilk-25">
+<head>
+    <meta charset="utf-8"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0"/>
+    @vite('frontend/js/app.js')
+</head>
+<body class="h-full">
+<div id="app"></div>
+</body>
+</html>
```
Comment (kamilpiech97) [inline frontend/views/app.blade.php:11]:
```suggestion
</html>

```

File: public/api/faculties/faculties.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        "id": 1,
+        "name": "Wydział Nauk Społecznych i Humanistycznych",
+        "directory": "wydzial-humanistyczny"
+    },
+    {
+        "id": 2,
+        "name": "Wydział Nauk o Zdrowiu i Kulturze Fizycznej",
+        "directory": "wydzial-zdrowie-kultura"
+    }
+]
```
Comment (kamilpiech97) [inline public/api/faculties/faculties.json:17]:
```suggestion
]

```
some `json` files has the same problem.

File: public/api/faculties/wydzial-humanistyczny/companies.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-humanistyczny/companies.json:1]:
```suggestion
[]

```

File: public/api/faculties/wydzial-humanistyczny/documents/documents.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "id": 0,
+        "displayedName": "Testowy plik",
+        "filePath": "example.txt"
+    }
+]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-humanistyczny/documents/documents.json:7]:
```suggestion
]

```

File: public/api/faculties/wydzial-humanistyczny/filters.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "country": [],
+    "city": [],
+    "specialization": [],
+    "tags": []
+}
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-humanistyczny/filters.json:6]:
```suggestion
}

```

File: public/api/faculties/wydzial-techniczny/companies.json [hunk 6]
```diff
@@ -0,0 +1,2690 @@
...
+        "email": "praktyki.centralazus@zus.pl",
+        "phoneNumber": "(22) 560 16 00",
+        "isPaid": "NIE",
+        "logoFile": "",
+        "socialMedia": {
+            "facebook": "",
+            "linkedIn": ""
+        }
+    }
+]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-techniczny/companies.json:2690]:
```suggestion
]

```

File: public/api/faculties/wydzial-techniczny/documents/documents.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "id": 0,
+        "displayedName": "Testowy plik",
+        "filePath": "example.txt"
+    }
+]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-techniczny/documents/documents.json:7]:
```suggestion
]

```

File: public/api/faculties/wydzial-techniczny/filters.json [hunk 8]
```diff
@@ -0,0 +1,702 @@
...
+        {
+            "id": 15,
+            "name": "hosting",
+            "searchMatchIds": [
+                52
+            ],
+            "childrenIds": []
+        }
+    ]
+}
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-techniczny/filters.json:702]:
```suggestion
}

```

File: public/api/faculties/wydzial-zdrowie-kultura/companies.json [hunk 9]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-zdrowie-kultura/companies.json:1]:
```suggestion
[]

```

File: public/api/faculties/wydzial-zdrowie-kultura/documents/documents.json [hunk 10]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "id": 0,
+        "displayedName": "Testowy plik",
+        "filePath": "example.txt"
+    }
+]
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-zdrowie-kultura/documents/documents.json:7]:
```suggestion
]

```

File: public/api/faculties/wydzial-zdrowie-kultura/filters.json [hunk 11]
```diff
@@ -0,0 +1,6 @@
+{
+    "country": [],
+    "city": [],
+    "specialization": [],
+    "tags": []
+}
```
Comment (EwelinaLasowy) [inline public/api/faculties/wydzial-zdrowie-kultura/filters.json:6]:
```suggestion
}

```

File: backend/storage/logs/.gitignore [hunk 12]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (EwelinaLasowy) [inline backend/storage/logs/.gitignore:2]:
```suggestion
!.gitignore

```

File: docker-compose.yml [hunk 13]
```diff
@@ -2,44 +2,104 @@ version: "3.8"
...
+
+networks:
+  intern-dev:
+    driver: bridge
+
+volumes:
+  intern-mysql-data:
+    name: intern-mysql-data
+  intern-redis-data:
+    name: intern-redis-data
```
Comment (EwelinaLasowy) [inline docker-compose.yml:105]:
```suggestion
    name: intern-redis-data
    
```

**Review by EwelinaLasowy — COMMENTED** [review]
Think about getting rid of comments. Also please check if all dependencies are up-to-date.

File: public/api/faculties/faculties.json (hunk 2 above)
Comment (plencka) [reply public/api/faculties/faculties.json]: That's right, I'll update the script that generates those.

**Review by MatArasim — COMMENTED** [review]
[mistakes were made]

**Review by EwelinaLasowy — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**